homeworks/lesia.turua_lesiaturuta/PokemonProject/scripts/pokemon.js

```diff
@@ -309,58 +66,138 @@ const renderCards = (cardContainer, pokemonsArray) => {
+        .toLocaleLowerCase()
+        .includes(text.toLocaleLowerCase()));
+
+const getPokemons = async (urlPokemons) => {
+    try {
+        const response = await fetch(urlPokemons);
+        return await response.json();
+    } catch (e) {
+        alert('Error url, please update this page');
+        console.error(e);
+        return [];
     }
 };
 
-const filterPokemons = (checkBig, checkSmall) => {
-    let newPokemons = pokemons;
-    if (checkBig && checkSmall) {
-        newPokemons = pokemons.filter((pokemon) => pokemon.weight < 100 && pokemon.weight > 50);
-    } else if (checkBig && !checkSmall) {
-        newPokemons = pokemons.filter((pokemon) => pokemon.weight > 100);
-    } else if (!checkBig && checkSmall) {
-        newPokemons = pokemons.filter((pokemon) => pokemon.weight < 50);
+const getFilterPokemons = () => {
+    let newPokemons = [];
+    if (searchValue && typeValue) {
+        newPokemons = getPokemonsByValet(searchValue, pokemons);
+        newPokemons = newPokemons.length
+            ? getPokemonsByType(typeValue, newPokemons)
+            : getPokemonsByType(typeValue, pokemons);
+    } else if (searchValue && !typeValue) {
+        newPokemons = getPokemonsByValet(searchValue, pokemons);
+    } else if (!searchValue && typeValue) {
+        newPokemons = getPokemonsByType(typeValue, pokemons);
+    } else if (!typeValue && !searchValue) {
+        newPokemons = pokemons;
+    }
+
+    if (checkboxBig && checkboxSmall && newPokemons.length) {
+        newPokemons = newPokemons.filter((pokemon) => pokemon.weight < 100 && pokemon.weight > 50);
+    } else if (checkboxBig && !checkboxSmall && newPokemons.length) {
+        newPokemons = newPokemons.filter((pokemon) => pokemon.weight > 100);
+    } else if (!checkboxBig && checkboxSmall && newPokemons.length) {
+        newPokemons = newPokemons.filter((pokemon) => pokemon.weight < 50);
+    }
+
+    if (highFilterEl.checked && newPokemons.length) {
+        newPokemons = sortPokemonsDown(newPokemons);
+    }
+    if (lowFilterEl.checked && newPokemons.length) {
+        newPokemons = sortPokemonsUp(newPokemons);
     }
+    typesPokemons = updateTypes(newPokemons);
+    renderSelect(typeValue);
     return newPokemons;
 };
 
-const init = () => {
-// disabled for html elements
-    selectSearchEl.disabled = true;
-    selectSearchRadioOnlyNewEl.disabled = true;
-    selectSearchRadioFavoritesEl.disabled = true;
-    // selectSearchInputSearchEl.disabled = true;
-    // selectSearchInputButtonEl.disabled = true;
+const init = async () => {
+    try {
+        pokemons = await getPokemons(url);
+    } catch (e) {
+        alert('Error url, please update this page');
+        console.error(e);
+    }
+    typesPokemons = updateTypes(pokemons);
+    renderSelect(typeValue);
+    loadingEl.classList.add('display-none');
 
-    selectSearchCheckboxBigEl.addEventListener('change', (e) => {
+    checkboxBigEl.addEventListener('change', (e) => {
```

Comment:
I still think, that `init` function is too long.
Could you make a `addHandlers` function and move to it all logic related to `addEventListener`

Reply:
fixed